### ISSUE TITLE
Waiting for Lean server to start confusing message

### ISSUE THREAD

**Issue by PatrickMassot** [leanprover/vscode-lean4#323]
When the extension completely fails to find a project, it [displays](https://github.com/leanprover/vscode-lean4/blob/a8659be9548b62ed74ac36e43f5c73806f3955c8/lean4-infoview/src/infoview/main.tsx#L59) "Waiting for Lean server to start" forever.

Steps to reproduce:
* Create folder `/tmp/not_a_project`
* Create file `/tmp/not_a_project/test.lean` containing Lean code, say `#check Nat`
* Open folder `/tmp/not_a_project` in VSCode

The infoview opens and says "Waiting for Lean server to start..." forever. The simplest improvement would be to add a paragraph starting with "If this messages stays for more than 20 seconds then you may want to..."

**Comment by mhuisi** [comment]
Another instance of this with a different cause can be found [on the Lean Zulip](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/Installation.3A.20Interesting.20elan.20errors/near/391723961).

**Comment by PatrickMassot** [comment]
I now see an error message but not the one you quote. And the one I see does not really give instructions on how to improve things.
![image](https://github.com/leanprover/vscode-lean4/assets/14060883/e091547c-7851-419f-a283-55b0a0a97eab)

**Comment by mhuisi** [comment]
You have no default toolchain set, correct?

> I now see an error message but not the one you quote. And the one I see does not really give instructions on how to improve things. ![image](https://user-images.githubusercontent.com/14060883/278390413-e091547c-7851-419f-a283-55b0a0a97eab.png)

Can you click the notification bell at the bottom right? When there are too many notifications, VS Code likes to hide them there.

**Comment by mhuisi** [comment]
@PatrickMassot With the same setup as before you should now see two notifications with #347 and 0.0.118: One informing you that you lack a default toolchain, and one informing you that you should open a folder.

**Comment by PatrickMassot** [comment]
Indeed, this what I see. This is an improvement, but still not maximally helpful. It states what is wrong but gives only a little bit of help about what to do next. 

The message about not being in a project could contain a button proposing to open the Setup guide.

The message about not having a Lean toolchain contains no explanation or pointer to explanation about what the user should do. Do we want a button the sets up a default toolchain? A button proposing to open the Setup guide?

**Comment by mhuisi** [comment]
The default toolchain error should be pretty rare because it only occurs if the user both fiddled with `elan toolchain uninstall` and opened the wrong folder. I'd hope the amount of users that know how to use elan on the command line but do not know what a Lean project folder is to be relatively small.

In the future, we want better diagnostics for broken setups in general (old versions, missing dependencies, broken lean installations, etc.) and a VS Code frontend for elan.

> The message about not being in a project could contain a button proposing to open the Setup guide.

This is a good suggestion for all errors that new users are likely to face in general. I'll keep it in mind.

**Comment by PatrickMassot** [comment]
Ok feel free to close the issue if you are happy with the current state (which is already very good indeed!).
